relay-metrics/src/protocol.rs

```diff
@@ -163,88 +164,225 @@ fn is_valid_name(name: &str) -> bool {
     false
 }
 
-/// The namespace of a metric.
-///
-/// Namespaces allow to identify the product entity that the metric got extracted from, and/or
-/// identify the use case that the metric belongs to. These namespaces cannot be defined freely,
-/// instead they are defined by Sentry. Over time, there will be more namespaces as we introduce
-/// new metrics-based products.
-///
-/// Right now this successfully deserializes any kind of string, but in reality only `"sessions"`
-/// (for release health) and `"transactions"` (for metrics-enhanced performance) is supported.
-/// Everything else is dropped both in the metrics aggregator and in the store service.
-#[derive(Clone, Copy, Debug, PartialEq, Eq)]
-pub enum MetricNamespace {
-    /// Metrics extracted from sessions.
-    Sessions,
-    /// Metrics extracted from transaction events.
-    Transactions,
-    /// Metrics that relay either doesn't know or recognize the namespace of, will be dropped before
-    /// aggregating. For instance, an MRI of `c:something_new/foo@none` has the namespace
-    /// `something_new`, but as Relay doesn't support that namespace, it gets deserialized into
-    /// this variant.
-    ///
-    /// Relay currently drops all metrics whose namespace ends up being deserialized as
-    /// `unsupported`. We may revise that in the future.
-    Unsupported,
+/// Enumerates the most common session-names.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum SessionsKind {
+    Session,
+    User,
+    Error,
 }
 
-impl std::str::FromStr for MetricNamespace {
-    type Err = ParseMetricError;
+impl Display for SessionsKind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Session => write!(f, "session"),
+            Self::User => write!(f, "user"),
+            Self::Error => write!(f, "error"),
+        }
+    }
+}
 
-    fn from_str(ns: &str) -> Result<Self, Self::Err> {
-        match ns {
-            "sessions" => Ok(MetricNamespace::Sessions),
-            "transactions" => Ok(MetricNamespace::Transactions),
-            _ => Ok(MetricNamespace::Unsupported),
+/// Enumerates the most common transaction-names, with a catch-all for any other names.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum TransactionsKind<'a> {
+    User,
+    Duration(DurationUnit),
+    CountPerRootProject,
+    Breakdowns(&'a str),
+    Measurements {
+        kind: Measurementkind<'a>,
+        unit: MetricUnit,
+    },
+    Custom {
```

Comment:
Is this variant instantiated anywhere? I think this should move into `MeasurementKind`. That is, there are custom _measurements_, but not fully custom transaction metrics.